wandb/sdk/data_types/utils.py

```diff
@@ -92,10 +94,28 @@ def val_to_json(
 
             items = _prune_max_seq(val)
 
-            for i, item in enumerate(items):
-                item.bind_to_run(
-                    run, key, namespace, id_=i, ignore_copy_err=ignore_copy_err
-                )
+            if _server_accepts_image_filenames():
+                for item in items:
+                    item.bind_to_run(
+                        run=run,
+                        key=key,
+                        step=namespace,
+                        ignore_copy_err=ignore_copy_err,
+                    )
+            else:
+                for i, item in enumerate(items):
+                    item.bind_to_run(
+                        run=run,
+                        key=key,
+                        step=namespace,
+                        id_=i,
```

Comment:
We may have discussed this, but perhaps you can refresh me. In the mp setting, is there any chance of a collision between processes logging the same key here?

Reply:
yeah exactly... it is possible since the step could be the same (it is the user space step) and if these process are same length it would cause a namespace collision:
in the below example all workers will have their own user space step and same length of image sequence hence collision
```
def worker(run):
    run.log("images": [Image(img) for img in images])
    
 def main():
     run = wandb.init()
     processes = [Process(target=worker, args=(run,)) for _ in range(num_processes)]
     for p in processes:
        p.start()
        
     for p in processes:
        p.join()
      run.finish()
if __name__ == "__main__":
    main()
```